### DISCUSSION TITLE
Pull request format discussion

### DISCUSSION THREAD

**Discussion by IcePigZDB** [CDDSCLab/Weekly-Group-Meeting-Paper-List#31] [Ideas]
# PR format discussion

discussion are welcome.

### Summary of talk

Date:

Topic:

Keywords:
### Format check List
- [x] Conforming to English specifications especially in `README.md`.
- Capitalize the first letter.
- Use English punctuation with appropriate space.
- ...
- [x] Remember to add links to reference papers.
- [x] Make sure the link is accessible.
- Use `-` to connect filename rather than ` `, the latter will casue dead link.
- [x] Make sure latex formulas display normally, more can be found in [#31](https://github.com/CDDSCLab/Weekly-Group-Meeting-Paper-List/discussions/31).
- [x] Pass the markdown lint check, more can be found in [#31](https://github.com/CDDSCLab/Weekly-Group-Meeting-Paper-List/discussions/31#discussioncomment-3828772).

### Dispaly latex formulas in github flavored markdown

1. Github has supported mathematical expressions in markdown , more information please refer to this [blog](https://github.blog/2022-05-19-math-support-in-markdown/). 🎉🎉🎉

2. [readme2tex](https://github.com/leegao/readme2tex).

3. [rendor latex formula to image with Github official service](https://gist.github.com/a-rodin/fef3f543412d6e1ec5b6cf55bf197d7b).
`https://render.githubusercontent.com/render/math?math=`

    - Vscode extension [math to image](https://marketplace.visualstudio.com/items?itemName=MeowTeam.vscode-math-to-image).

    - Use a scripts, a simple example in python, style ref is [math to image](https://marketplace.visualstudio.com/items?itemName=MeowTeam.vscode-math-to-image), code [ref repo](https://github.com/Crysple/MarkZhiihuDown/tree/master).
    Save as python script and run `python3 script-filename.py target-markdown-filename.md`.
    ```python
	import sys
	import re
	from urllib.parse import quote
	# $$
	# formula
	# $$
	interline_tag = '\n<img src="https://render.githubusercontent.com/render/math?math={}" class="ee_img tr_noresize" eeimg="1">\n'
	interline_pattern = '\$\$\n*((.|\n)*?)\n*\$\$' # (.|\n) . or \n
	# $formula$
	inline_tag = '<img src="https://render.githubusercontent.com/render/math?math={}" class="ee_img tr_noresize" eeimg="1">'
	inline_pattern = '\$\n*(.*?)\n*\$'

	def replace_tex(content):
		def dashrepl_quote(matchobj, tag):
			formular = matchobj.group(1)
	                # quote & to %26 etc in $$ x&= a+1$$, otherwise, thers will be error when render with githubusercontent.com
			formular = quote(formular) 
			return tag.format(formular, formular)

		content = re.sub(interline_pattern, lambda mo: dashrepl_quote(mo, interline_tag), content)
		content = re.sub(inline_pattern, lambda mo: dashrepl_quote(mo, inline_tag), content)

		return content

	if __name__=='__main__':
		assert len(sys.argv) > 1, "Error: need filename as a argument"
		filename = sys.argv[1]
		with open(filename, mode='r',encoding='utf-8') as f:
			content = f.read()
		with open(filename, mode='w',encoding='utf-8') as f:
			f.write(replace_tex(content))
	```

	Rendering multi-line formula example with githubusercontent:

      ```latex
      $$
      \begin{align}
      x = v_0\cos\theta t\\
      y = v_0\sin\theta t - \frac{1}{2}gt^2
      \end{align}
      $$
      ```

	<img src="https://render.githubusercontent.com/render/math?math=%5Cbegin%7Balign%7D%0Ax%20%3D%20v_0%5Ccos%5Ctheta%20t%5C%5C%0Ay%20%3D%20v_0%5Csin%5Ctheta%20t%20-%20%5Cfrac%7B1%7D%7B2%7Dgt%5E2%0A%5Cend%7Balign%7D" class="ee_img tr_noresize" eeimg="1">

      ```latex
      $$
      \begin{align}
      x &= v_0\cos\theta t\\
      y &= v_0\sin\theta t - \frac{1}{2}gt^2
      \end{align}
      $$
      ```

	<img src="https://render.githubusercontent.com/render/math?math=%5Cbegin%7Balign%7D%0Ax%20%26%3D%20v_0%5Ccos%5Ctheta%20t%5C%5C%0Ay%20%26%3D%20v_0%5Csin%5Ctheta%20t%20-%20%5Cfrac%7B1%7D%7B2%7Dgt%5E2%0A%5Cend%7Balign%7D" class="ee_img tr_noresize" eeimg="1">

      ```latex
      $$
      x = v_0\cos\theta t\\
      y = v_0\sin\theta t - \frac{1}{2}gt^2
      $$
      ```

	<img src="https://render.githubusercontent.com/render/math?math=x%20%3D%20v_0%5Ccos%5Ctheta%20t%5C%5C%0Ay%20%3D%20v_0%5Csin%5Ctheta%20t%20-%20%5Cfrac%7B1%7D%7B2%7Dgt%5E2" class="ee_img tr_noresize" eeimg="1">

    Please use `\begin{align}` to display in rows.
    More discussion in [stackoveflow](https://stackoverflow.com/questions/11256433/how-to-show-math-equations-in-general-githubs-markdownnot-githubs-blog).

**Comment by ehds** [comment]
Github has supported mathematical expressions in markdown , more information please refer to this [blog](https://github.blog/2022-05-19-math-support-in-markdown/).  🎉🎉🎉
Inline example:  mass-energy equation $E=mc^2$

multi-line example： Maxwell's equations

$$
 \begin{cases} \nabla \cdot \mathbf{E} = \frac{\rho}{\epsilon_0} \\ 
\nabla \times \mathbf{E} = -\frac{\partial}{\partial t} \mathbf{B} \\ 
\nabla \cdot \mathbf{B} = 0 \\ 
\nabla \times \mathbf{B} = \mu_0 \mathbf{J} + \mu_0 \epsilon_0 \frac{\partial}{\partial t} \mathbf{E} \\ 
\end{cases}
$$

**Comment by IcePigZDB** [comment]
# Markdown lint
Markdown lint check for new PR has ben introduced by #41, #43 . 

To make it easier for you to pass the check, here is a possible way.

- Write your mardown file on vscode with two plugins:
    * [Markdown All In Ene](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one#review-details)
    * [markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint)

- Format markdonw file by vscode format api according to [vscode doc](https://code.visualstudio.com/docs/editor/codebasics#_formatting).

More approaches are welcome :blush:.
